### ISSUE TITLE
[Caffe2] Switch to eigenteam mirror for eigen

### ISSUE THREAD

**Issue by xkszltl** [pytorch/pytorch#7089]
## Issue description

Seems that Eigen now have an official git mirror: `https://github.com/eigenteam/eigen-git-mirror.git`
Should we switch to there rather than using RLovelett repo?

## Code example

```
[submodule "third_party/eigen"]
	path = third_party/eigen
	url = https://github.com/RLovelett/eigen.git
```

**Comment by xkszltl** [comment]
PR created #7090

**Comment by kkdai** [comment]
Hi, 
Original repo https://github.com/RLovelett/eigen.git has been removed and introduce issue #7793.
Could we make new tag 0.4.0 to fix #7793?
